### PR TITLE
Expose Horizontal Pod Autoscaler Behavior and add hpa scaledown test

### PR DESCRIPTION
**Pull request by kevinearls**

### CODE REVIEW

File: apis/v1alpha1/opentelemetrycollector_types.go [hunk 1]
```diff
@@ -40,6 +40,13 @@ type OpenTelemetryCollectorSpec struct {
 	// MaxReplicas sets an upper bound to the autoscaling feature. If MaxReplicas is set autoscaling is enabled.
 	// +optional
 	MaxReplicas *int32 `json:"maxReplicas,omitempty"`
+
+	// Autoscaler specifies the pod autoscaling configuration to use
+	// for the OpenTelemetryCollector workload.
+	//
+	// +optional
+	Autoscaler *AutoscalerSpec `json:"autoscaler,omitempty"`
```
Comment (jaronoff97) [inline apis/v1alpha1/opentelemetrycollector_types.go:48]: question: did we consider embedding the HPA spec in here? Or at least embed the [autoscaling behavior spec](https://pkg.go.dev/k8s.io/api/autoscaling/v2beta2#HorizontalPodAutoscalerBehavior) here?

File: apis/v1alpha1/opentelemetrycollector_webhook.go [hunk 2]
```diff
@@ -129,6 +129,15 @@ func (r *OpenTelemetryCollector) validateCRDSpec() error {
 			return fmt.Errorf("the OpenTelemetry Spec autoscale configuration is incorrect, minReplicas should be one or more")
 		}
 
+		if r.Spec.Autoscaler != nil {
```
Comment (jaronoff97) [inline apis/v1alpha1/opentelemetrycollector_webhook.go:132]: We should also check that min or max replicas is set in order to use this feature

**Review by iblancasa — APPROVED** [review]

File: apis/v1alpha1/opentelemetrycollector_webhook.go (hunk 2 above)
Comment (kevinearls) [reply apis/v1alpha1/opentelemetrycollector_webhook.go:132]: That's inside an if statement that's checking whether maxReplicas is set.

File: apis/v1alpha1/opentelemetrycollector_types.go (hunk 1 above)
Comment (kevinearls) [reply apis/v1alpha1/opentelemetrycollector_types.go:48]: I was just adding enough code to be able to get an e2e test to work within the time allocated, which means we need to scale down much quicker that the default 300 seconds.

@pavolloffay what do you think?  Do I need to add the other values of PA scaling rules here?

File: tests/e2e/autoscale/00-assert.yaml [hunk 3]
```diff
@@ -13,4 +13,7 @@ metadata:
 spec:
   minReplicas: 1
   maxReplicas: 2
+# This is not neccesarily exact.  We really just want to wait until this is no longer <unknown>
```
Comment (iblancasa) [inline tests/e2e/autoscale/00-assert.yaml:16]: This is a really good "hack"

**Review by iblancasa — APPROVED** [review]

**Comment by pavolloffay** [comment]
Does this resolve #942 ?

**Comment by kevinearls** [comment]
@pavolloffay Yes

File: apis/v1alpha1/opentelemetrycollector_types.go (hunk 1 above)
Comment (jaronoff97) [reply apis/v1alpha1/opentelemetrycollector_types.go:48]: For my use case, I know that I would like to be able to specify policies and not just `StabilizationWindowSeconds`. Embedding only `StabilizationWindowSeconds` is going to make it so we need to add in each feature on request, making a code change for each one.

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**